### PR TITLE
Create test_build.yml

### PR DESCRIPTION
**Pull request by ankitade**
Summary:
Add a test_build.yml to build wheel, test install it and upload to GHA. This is the first step toward setting up our release CI. 
Next step would be to  install torch* dependencies and run the tests

Test plan:
- This workflow should be manually triggerable but cant test that till the merge happens 
- So created a PR with the workflow trigger as PR temporarily --> Run on the PR https://github.com/facebookresearch/multimodal/actions/runs/2405089199
Able to download the wheel

### CODE REVIEW

**Comment by langong347** [comment]
>Next step would be to install torch* dependencies and run the tests

Are you referring to the upstreams manually installed by conda before running `python setup.py`?
We will need to feed them into `install_requires` so `setuptools` will download them from PyPI .

**Comment by ankitade** [comment]
- I tried incorporating the torch* dependencies into setup.py but there is no principle way of doing it for nightlies. So it is still manual installation for them. So the testing point was about doing a pip install for those libs and running pytest (after installing the torchmultimodal wheel step)
-  Why would we want to download the wheel? At some point, we should just add publishing to pypi and conda steps to it?

**Comment by langong347** [comment]
> I tried incorporating the torch* dependencies into setup.py but there is no principle way of doing it for nightlies. 

Have you tried telling `setuptools` to `install_require` the name of the upstream like [this](https://github.com/pytorch/vision/blob/main/setup.py#L56-L65)? Will it just pick whatever the most recent stable version from PyPI for the upstream? Except for when we want to pin the upstream version (such as nightly for CI), an arbitrary end user does not need to care and we should remove the extra step for them. What do you think?


> Why would we want to download the wheel? 
To share the distribution archive between jobs, e.g.,
- Build: produce the archive and upload the artifacts
- Test the build: download the archive and test import of the library
- Unit test: download the archive and run unit tests
- Distribution (PyPI/conda): download the archive, set up credentials and upload to PyPI/Conda

I suggest we add "test the build" to the current build workflow as otherwise it lacks the validation.
As a stretch (or file a BE task), we can include the download step in the unit test workflow.
Distribution to PyPI/Conda will need to wait until we move to PyTorch org to leverage the unified channel.

For reference here are PyTorch's workflows for pull requests: 
- [build](https://github.com/pytorch/pytorch/blob/master/.github/workflows/_linux-build.yml)
- [unit test](https://github.com/pytorch/pytorch/blob/master/.github/workflows/_linux-test.yml)
For [nightlies](https://github.com/pytorch/pytorch/blob/master/.github/workflows/generated-linux-binary-manywheel-nightly.yml): build, test the build, upload.

File: .github/workflows/test_build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Build wheel
+        run: |
+          pip3 install wheel
+          python setup.py bdist_wheel
+      - name: Install wheel
+        run: |
+          pip3 install dist/torchmultimodal*.whl
+      - name: Upload to github
```
Comment (langong347) [inline .github/workflows/test_build.yml:25]: I would remove this step for now as there's no other CI job that needs to share the uploaded artifacts.

**Review by langong347 — APPROVED** [review]

File: .github/workflows/test_build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Test build workflow
```
[the remark this answers is not in the document]
Comment (ankitade) [reply .github/workflows/test_build.yml:1]: yes we should eventually do that. for now, leaving it as is till we test it out and add the release capability

**Pull request closed without merging**